src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -373,6 +320,62 @@ public void applyEstimatedTimetable(
       // Always release lock
       bufferLock.unlock();
     }
+    return UpdateResult.ofOptions(results);
+  }
+
+  private List<Optional<UpdateError>> apply(
```

Comment:
Based on the previous discussions this week, I think we should avoid types like `List<Optional<?>>` Instead we can process the counts already here, or we can return `List<Result<?, UpdateError>>`

Reply:
So was the consensus that we should never use Optional, even for return types? What benefit does it have to use another type which has an a second, unused type parameter?

Reply:
I've implemented your request in fc1d42f6afe5f04197b6a0231d386b215fb43d70. Can you check if you like this better?